File: server_Socket/src/main/scala/serverSocketWork.scala

import java.io.{BufferedReader, InputStreamReader, PrintWriter}
import java.net.Socket
import javax.net.ssl.{HttpsURLConnection, SSLSocketFactory}

case class serverSocketWork(client:Socket) extends Runnable{
  override def run(): Unit = {
    val in=new BufferedReader(new InputStreamReader(client.getInputStream))
    val out=new PrintWriter(client.getOutputStream, true)
    //def getHTTPRequest() = {
   /*   val requestLine = in.readLine()
      val (method, url, version) =
        (requestLine.takeWhile(_ != ' '), requestLine.dropWhile(_ != ' ').takeWhile(_ != ' '),
          requestLine.dropWhile(_ != ' ').dropWhile(_ != ' ').takeWhile(_ != '\r'))
      val headerTitle = in.readLine()
      val (headerFieldName, value)=(headerTitle.takeWhile(_!=' '), headerTitle.dropWhile(_!=' ').takeWhile(_!='\r'))
      var headerBody:String=""
      var line:String=null
      while ((!(line=in.readLine()).equals(headerTitle))) {
        headerBody+=line
      }
      line=in.readLine()
      var entityBody:String=""
      while ((line=in.readLine())!=null) {
          entityBody+=line
      }*/
   // }
    val https=HttpsURLConnection

  }
}

File: myWork/serverSocketWork.java
package myWork;


import javax.net.ssl.HttpsURLConnection;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.Socket;
import java.net.URL;
import java.util.*;

public class serverSocketWork implements Runnable {
    private Socket client;
    private Boolean executed=false;

    public serverSocketWork(Socket client) {
        this.client = client;
    }

    @Override
    public void run() {
        if (executed) return;
        executed=true;
        BufferedReader in;
        try {
            InputStream is=client.getInputStream();
            in = new BufferedReader(new InputStreamReader(is));
            String line=in.readLine();
            URL url = new URL();
            HttpsURLConnection con = (HttpsURLConnection) url.openConnection();
        } catch (IOException e) {
            throw new RuntimeException("Error in creating a buffered reader");
        }

    }
}
